eventlet/support/afinet.py

```diff
@@ -0,0 +1,46 @@
+"""
+Test availabilty of AF_INET6 on a machine
+
+Uses socket API to determine configuration
+"""
+
+from socket import *
+
+def check_ipv6_lo_addr():
+    try:
+        sock = socket(AF_INET6, SOCK_STREAM)
+        sock.bind(('::1', 0))
+        return True
+    except:
+        return False
+
+def check_ipv6_hosts_file():
+    if not check_ipv6_lo_addr():
+        return False
+    addrinfo = getaddrinfo(gethostname(), 0)
+    if (addrinfo[0][0] == AF_INET6):
+        return True
+    return False
+
+core_use_ipv6_lo_addr = check_ipv6_lo_addr()
+use_ipv6_by_default = check_ipv6_hosts_file()
+
+ip_defaults = {}
```

Comment:
Absolutely no reason to use mutable, string indexed dictionary here. Go with namedtuple or object with cached lazy property.